CHANGES.md

```diff
@@ -1,6 +1,421 @@
+
+Third-Party Software Changes
+----------------------------
+* updated the bundled version of Qthreads to version 1.17
+* updated FLTK version from 1.3.5 to 1.3.8 for improved support on Mac M1
+* updated the Python package versions used by 'chpldoc' as follows:
+  - Babel: version 2.10.3
+  - Jinja2: version 3.1.2
+  - MarkupSafe: version 2.1.1
+  - Pygments: version 2.12.0
+  - Sphinx: version 4.5.0
+
+Developer-oriented changes: Process
+-----------------------------------
+
+Developer-oriented changes: Documentation
+-----------------------------------------
+
+Developer-oriented changes: Naming Changes
+------------------------------------------
+
+Developer-oriented changes: Module changes
+------------------------------------------
+* replaced hand-written instability warnings with the `@unstable` annotation
+
+Developer-oriented changes: Performance improvements
+----------------------------------------------------
+
+Developer-oriented changes: Makefile / Build-time changes
+---------------------------------------------------------
+* turned off LLVM assertions for `CHPL_DEVELOPER` builds
+* adjusted generation of `chpl-env-gen.h` to better handle special characters
+
+Developer-oriented changes: Compiler Flags
+------------------------------------------
+* updated `--dyno` to activate the 'dyno' scope resolver
+* added `--warn-int-uint` to warn about implicit `int` to `uint` conversions
+* added `--report-gpu-transform-time` to measure time for GPU transformations
+* added `--library-cmakelists` to generate `cmake` files for a Chapel library
+
+Developer-oriented changes: Compiler improvements/changes
+---------------------------------------------------------
+* removed support for the old, pre-'dyno' parser
+
+Developer-oriented changes: 'dyno' Compiler improvements/changes
+----------------------------------------------------------------
+* migrated many semantic checks from the production parser to 'dyno'
+* migrated implicit module warnings to 'dyno'
+* made numerous improvements to 'dyno's scope resolution capabilities
+  - added support for `this.` and `super.` in `use`/`import` statements
+  - added support for `include` statements
+  - added support for task intents and reduce intents
+  - added support for try-catch statements
+* added conversions from uAST to resolved `SymExpr`s in the production compiler
+* made numerous improvements to 'dyno's type/call resolution capabilities
+  - added support for resolving enums, conditional expressions, and ranges
+  - added support for loop index variables and param for-loops
+  - added support for varargs argument lists
+  - added support for evaluating `==` and `!=` on types
+  - added support for type unification on routines with multiple `return`s
+  - added support for resolving declarations with both a type and initializer
+  - added ability to reject less valid calls
+  - improved resolution's ability to reject more invalid calls
```

Comment:
I don't know what this one is but it seems a bit nebulous and maybe we can leave it out

Reply:
Oops, I missed this earlier (so preserved it in some form in my latest refactor).  I interpreted this as having resolution filter out candidates that were not good matches (and tried to reword it that way in my latest set of changes).  I believe this came from @DanilaFe's contributions (and I may have made it into two bullets incorrectly in my attempt to find a good wording for it?  Or maybe someone else proposed a similar bullet?)  My current attempt at wording is:

>   - improved support for rejecting invalid overload candidates

Reply:
Works for me if @DanilaFe is happy with it

Reply:
I'm happy with it!